lm75bd/lm75bd.c

```diff
@@ -27,7 +31,26 @@ error_code_t lm75bdInit(lm75bd_config_t *config) {
 
 error_code_t readTempLM75BD(uint8_t devAddr, float *temp) {
   /* Implement this driver function */
+  uint8_t tempReg[1]= {0};
+  i2cSendTo(LM75BD_OBC_I2C_ADDR, tempReg, 1);
+  uint8_t tempData[2] = {0,0};
+  i2cReceiveFrom(LM75BD_OBC_I2C_ADDR, tempData, 2);
+  int16_t tempBin = ((tempData[0] << 8)|(tempData[1]))>>5;
+  int16_t tempDec = 0;
+  BOOL negative = FALSE;
   
+  if (tempBin & (1 << 10)){
+      tempBin = (~tempBin)+1;
+      negative = TRUE;
+  }
+  for (uint16_t i=1; i < (1 << 10); i <<= 1){
+      tempDec += !!(tempBin & i)*i;
+  }
+  if (negative){
+      tempDec *= -1;
+  }
+  //convert to Celsius
+  *temp = tempDec * 0.125;
```

Comment:
This logic is a bit more complicated than it needs to be. The register holds the 11-bit temperature data in 2s complement form already. 

So this conversion can be done like:
```
int16_t regVal = tempData[0] << 8 | tempData[1];
*temp = (float)(regVal >> 5) * 0.125
```